pkg/envinfo/storage.go

```diff
@@ -133,7 +129,7 @@ func (ei *EnvInfo) DeleteStorage(name string) error {
 
 // GetStorageMountPath gets the mount path of the storage with the given storage name
 func (ei *EnvInfo) GetStorageMountPath(storageName string) (string, error) {
-	return ei.devfileObj.Data.GetVolumeMountPath(storageName)
+	return ei.devfileObj.Data.GetVolumeMountPath(storageName, ei.componentSettings.Name)
```

Comment:
So not sure if `ei.componentSettings.Name` is correct here. 

`GetVolumeMountPath()` takes a mount name and container name because essentially devfile can have many container components and each container component can have mounts with the same name, so we need to specify which container component in the devfile do you want the mount name from..

Reply:
Thinking about this a bit more, the return value of `GetVolumeMountPath` should be an array because each container component can have multiple mounts on the same volume for a container component can have

```
components:
  - name: test2
    volume:
        size: 1Gi
  - name: test
    container:
        image: xyz
        volumeMounts:
           - name: test2
             path: /path1
           - name: test2
             path: /path2
```

so this is essentially a bug with devfile/library new func too

Reply:
Is there a bug on `devfile/library` repo to track this? I don't suppose I can fix this correctly without this being fixed there. Or can I? Argh! It sucks that I don't have much idea about the devfile side of odo code. Every interaction is making me feel lost. 😕

Reply:
Yeah as @maysunfaisal pointed out, it should return all the mount paths for a given storage name and return a map (with container names) or array of the paths. 

> So not sure if ei.componentSettings.Name is correct here.

It is incorrect. We need to first fetch all the containers, then fetch the mount path for each container and return an array  of the path or a map. Maybe for now, we can return the first path from the first container and handle the rest when it's fixed on the library's side.

Reply:
@mik-dass right now the func `GetVolumeMountPath()` takes the mount name and the container name but returns the first path.. so essentially, to make this PR compatible with current odo master, @dharmit  would just need to pass the container name or know the container name to pass to this func.

@mik-dass do you want me to correct this on devfile/library master like you suggested? ie `GetVolumeMountPath` takes a mount name and returns a map of container -> mountPath or `GetVolumeMountPath` can take the mount name and container name and return an array of mountPaths for that container component

Reply:
Correct me if I'm wrong but FWIW, I just checked the repo and it looks like no one is using this func `(ei *EnvInfo) GetStorageMountPath(storageName string)`

Reply:
> current odo master, @dharmit would just need to pass the container name or know the container name to pass to this func.

Yes. In a later PR, we can update and fetch all the containers and call the function for all of them as it might be mounted at a different path in a different container. That would change the return signature of the function.

> @mik-dass do you want me to correct this on devfile/library master like you suggested? ie GetVolumeMountPath takes a mount name and returns a map of container -> mountPath or GetVolumeMountPath can take the mount name and container name and return an array of mountPaths for that container component

Yes that would be great :+1: The function can return an array of all the mountPaths for that given container component.

> Correct me if I'm wrong but FWIW, I just checked the repo and it looks like no one is using this func (ei *EnvInfo) GetStorageMountPath(storageName string)

It's being used here https://github.com/openshift/odo/blob/7e9ad7bda5c35d81890413a6524dff3ef73e1d94/pkg/odo/cli/storage/delete.go#L71

Reply:
PTAL https://github.com/openshift/odo/pull/4465/commits/5b494c927dfecfd5fa0bea683436356576d1d503

Reply:
looks good